builders/interact/ethereum-api/precompiles/call-permit.md

```diff
@@ -0,0 +1,275 @@
+
+The Call Permit Precompile is located at the following address:
+
+```text
+{{ networks.dancebox.precompiles.call_permit }}
+```
+
+!!! note
+    There can be some unintended consequences when using precompiles. Tanssi's Batch Precompile is derived from Moonbeam's, and as such, please familiarize yourself with [Moonbeam's Precompile Security Considerations](https://docs.moonbeam.network/builders/get-started/eth-compare/security/){target=\_blank}. 
+
+
+## The Call Permit Solidity Interface {: #the-call-permit-interface }
+
+[`CallPermit.sol`](https://github.com/moondance-labs/tanssi/blob/master/test/contracts/solidity/CallPermit.sol){target=\_blank} is a Solidity interface that allows developers to interact with the precompile's three methods.
+
+??? code "CallPermit.sol"
+
+    ```solidity
+    --8<-- 'code/builders/interact/ethereum-api/precompiles/call-permit/call-permit.sol'
+    ```
+
+The interface includes the following functions:
+
+???+ function "**dispatch**(*address* from, *address* to, *uint256* value, *bytes* data, *uint64[]* gaslimit, *uint256* deadline, *uint8* v, *bytes32* r, *bytes32* s) — dispatches a call on the behalf of another user with a EIP-712 permit. This function can be called by anyone or any smart contract. The transaction will revert if the permit is not valid or if the dispatched call reverts or errors (such as out of gas). If successful, the nonce of the signer is increased to prevent this permit to be replayed."
+
+    === "Parameters"
+
+        - `from` - the signer of the permit. The call will be dispatched on behalf of this address
+        - `to` - the address the call is made to
+        - `value` - the value being transferred from the `from` account
+        - `data` - the call data, or action to be executed
+        - `value` - the value being transferred from the `from` account
+        - `gasLimit` - the gas limit the dispatched call requires. Providing an argument for this parameter prevents the dispatcher from manipulating the gas limit
+        - `deadline` - the time in UNIX seconds after which the permit will no longer be valid. In JavaScript, you can get the current time in UNIX seconds by running `console.log(Date.now())` in a JavaScript script or a browser console
+        - `v` - the recovery ID of the signature. The last one byte of the concatenated signature
+        - `r` - the first 32 bytes of the concatenated signature
+        - `s` - the second 32 bytes of the concatenated signature
+
+
+??? function "**nonces**(*address* owner) — returns the current nonce for given owner"
+
+    === "Parameters"
+
+        - `owner` - the address of the account to check
+
+
+??? function "**DOMAIN_SEPARATOR**() — returns the EIP-712 domain separator which is used to avoid replay attacks. It follows the [EIP-2612](https://eips.ethereum.org/EIPS/eip-2612#specification){target=\_blank} implementation"
+
+    === "Parameters"
+
+        - This function takes no parameters
```

Comment:
you can just put None here instead, that has been the standard so far, and this would require a === "Returns" section